src/components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,96 @@
+import classNames from 'classnames';
+import { useMemo } from 'react';
+
+import { FilterBy } from '../../types/FilterBy';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  filterBy: FilterBy,
+  setFilterBy: (v: FilterBy) => void,
+  deleteTodo: (todoId: number) => void,
+};
+
+export const TodoFooter: React.FC<Props> = ({
+  todos,
+  filterBy,
+  setFilterBy,
+  deleteTodo,
+}) => {
+  // Hide the "Clear complited" if there isn't any completed todo
+  const someCompleted = useMemo(() => {
+    return todos.some(todo => todo.completed);
+  }, [todos]);
+
+  // Set filtering by "ALL, ACTIVE or COMPLITED"
+  const handleFiltering = (value: FilterBy) => {
+    if (value === filterBy) {
+      return;
+    }
+
+    setFilterBy(value);
+  };
+
+  // delete all complied todos
+  const deleteComplitedTodos = () => {
+    todos.map(todo => todo.completed && deleteTodo(todo.id));
```

Comment:
You should use 'forEach' instead of 'map' because 'map' is used to create a new array based on the result of the function. But here, you're not using the array that 'map' returns, so 'forEach' is more appropriate.